samples/bluetooth/peripheral_sc_only/src/main.c

```diff
@@ -102,9 +102,9 @@ static void pairing_complete(struct bt_conn *conn, bool bonded)
 	printk("Pairing Complete\n");
 }
 
-static void pairing_failed(struct bt_conn *conn)
+static void pairing_failed(struct bt_conn *conn, u8_t reason)
 {
-	printk("Pairing Failed. Disconnecting.\n");
+	printk("Pairing Failed (%#02x). Disconnecting.\n", reason);
```

Comment:
Id include a "0x" when printing the reason to indicate it is a hex value.

Reply:
That is what `#` does :)

Reply:
*mind blown* - I didn't realise there was a standard for this - I even had to go double check that the various libcs I use support this, and they do - minimal libc included. For consistency with the existing code base 0x%02x would be the right choice, but I think it does make sense to convert gradually the code base to use # and we have to start somewhere.

Reply:
summarising discussion from slack: we'll stick to the old style since using # produces inconsistent results due to exceptional handling of the value 0

Reply:
Have never seem that used before, yet it is documented in the man page of printf:

```
       #      The value should be converted to an "alternate form".  For o conversions, the first character of the output string is made zero (by  prefixing  a  0  if  it  was  not  zero
              already).   For  x  and  X  conversions,  a nonzero result has the string "0x" (or "0X" for X conversions) prepended to it.  For a, A, e, E, f, F, g, and G conversions, the
              result will always contain a decimal point, even if no digits follow it (normally, a decimal point appears in the results of those conversions only  if  a  digit  follows).
              For g and G conversions, trailing zeros are not removed from the result as they would otherwise be.  For other conversions, the result is undefined.
```
So yeah, we have been using 0x for ages without knowing # would do a better job converting the format.

Reply:
@Vudentz please check the backlog on #bluetooth on slack - we came to the conclusion that even though # is a standard it'll produce inconsistent results, i.e. we'll stick to explicitly specifying 0x for now